source/1-SDLC-organization/lib/cdk-bootstrap-template.yml

```diff
@@ -522,17 +522,6 @@ Resources:
               - sts:AssumeRole
             Resource: "*"
 
-            # AlloBoundaryDeletionForCICDSelfMutate
```

Comment:
Hmm removing this won't block self mutation ?

If we decide to go that way the adding codepipeline:StartPipelineExecution is useless and we need to remove slef. mutation in examples

Reply:
No, I was wrong.  

I thought it was required because my pipeline was throwing this error.  Turned out that it was watching the wrong branch/repo. That repo didn't have permissions boundary in the code, so self-mutation failed.   

This time I tested by getting everything green first.  Then start removing prod stage and rerun. Everything works. But removing the boundary will trigger access denied. The case that mutation fails is only when we remove permission boundary or the roles. In that case, they must use admin account to change the stack directly. If users just add/remove stage, it still works fine.

So I think it's good to keep the self-mutation. As long as users do not change role's permissions boundary. It works.